### PR TITLE
feat(azure-resources): VM Scale Set Network Interfaces

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/9258.

The API to list VM Scale Set Network Interfaces is on the `armnetwork` module and not on the `armcompute` module

### CODE REVIEW

**Review by hermanschaaf — APPROVED** [review]

File: plugins/source/azure/resources/services/compute/virtual_machine_scale_set_vm_network_interfaces.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		p, err := pager.NextPage(ctx)
+		if err != nil {
+			return err
+		}
+		res <- p.Value
+	}
+	return nil
+}
+
+func createVirtualMachineScaleSetsNetworkInterfaces(router *mux.Router) error {
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply plugins/source/azure/resources/services/compute/virtual_machine_scale_set_vm_network_interfaces.go:50]: This is a relation, so no need for a separate test file (that doesn't have a test 😄 ). We could put it in the top level table file, but then it's a bit disconnected from the actual API usage

[the remark this answers is not in the document]
Comment (hermanschaaf) [reply plugins/source/azure/resources/services/compute/virtual_machine_scale_set_vm_network_interfaces.go:50]: This is purely test code though right? Maybe this isn't the practice for the Azure plugin, but I usually create a `_test.go` file for the relation with functions like these even if the file doesn't contain any actual tests. I think test code should at least live in a test file, even if it is for the top-level table

[the remark this answers is not in the document]
Comment (erezrokah) [reply plugins/source/azure/resources/services/compute/virtual_machine_scale_set_vm_network_interfaces.go:50]: I don't have a strong opining so I can move it. I personally find it easier to write and review when it's in the same file

[the remark this answers is not in the document]
Comment (erezrokah) [reply plugins/source/azure/resources/services/compute/virtual_machine_scale_set_vm_network_interfaces.go]: Done in https://github.com/cloudquery/cloudquery/pull/9299/commits/f1cc15de204441ce7ed8e92d6bd10d2b210f7201

**Review by disq — APPROVED** [review]

**Pull request merged**